src/Util/src/FMI3/fmi3_enums.c

```diff
@@ -148,40 +148,43 @@ const char* fmi3_initial_to_string(fmi3_initial_enu_t ini) {
 }
 
 fmi3_initial_enu_t fmi3InitialDefaultsTable[fmi3_variability_enu_unknown][fmi3_causality_enu_unknown] = {
-    /*              parameter                  calculated parameter,        input                     output                       local                        independent */
-    /* constant */  {fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_unknown,  fmi3_initial_enu_exact,      fmi3_initial_enu_exact,      fmi3_initial_enu_unknown},
-    /* fixed   */   {fmi3_initial_enu_exact,   fmi3_initial_enu_calculated, fmi3_initial_enu_unknown,  fmi3_initial_enu_unknown,    fmi3_initial_enu_calculated, fmi3_initial_enu_unknown},
-    /* tunable */   {fmi3_initial_enu_exact,   fmi3_initial_enu_calculated, fmi3_initial_enu_unknown,  fmi3_initial_enu_unknown,    fmi3_initial_enu_calculated, fmi3_initial_enu_unknown},
-    /* discrete */  {fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_unknown,  fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown},
-    /* continuous */{fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_unknown,  fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown}
+    /*              parameter                  calculated parameter,        input                     output                       local                        independent               structural parameter*/
+    /* constant */  {fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_unknown, fmi3_initial_enu_exact,      fmi3_initial_enu_exact,      fmi3_initial_enu_unknown, fmi3_initial_enu_unknown},
+    /* fixed   */   {fmi3_initial_enu_exact,   fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_exact  },
+    /* tunable */   {fmi3_initial_enu_exact,   fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_exact  },
+    /* discrete */  {fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_exact  , fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown},
+    /* continuous */{fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_exact  , fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown}
 };
 
 FMILIB_EXPORT fmi3_variability_enu_t fmi3_get_default_valid_variability(fmi3_causality_enu_t c)
 {
-    if (c == fmi3_causality_enu_calculated_parameter || c == fmi3_causality_enu_parameter) {
-        return fmi3_variability_enu_fixed;
-    } else {
-        return fmi3_variability_enu_continuous;
+    switch(c){
```

Comment:
opted for a straight forward switch case, in particular to also address the causality = unknown case.